### PR TITLE
fix(OnyxDialog): prevent overflow and add z-index

### PR DESCRIPTION
**Pull request by larsrickert**
Relates to [#1241](https://github.com/SchwarzIT/onyx/issues/1241#issuecomment-2167831241)

Prevent overflow of the dialog content.

## Checklist

- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxDialog/OnyxDialog.ct.tsx [hunk 1]
```diff
@@ -34,6 +34,23 @@ test("should render in modal", async ({ mount, makeAxeBuilder, page }) => {
...
+  await mount(
+    <OnyxDialog label="Label" open>
+      {"Test".repeat(64)}
+    </OnyxDialog>,
+  );
+
+  // accessibility tests
+  const accessibilityScanResults = await makeAxeBuilder()
+    // the interactive/focusable content is provided by the project
+    .disableRules(["scrollable-region-focusable"])
```
Comment (BoppLi) [inline packages/sit-onyx/src/components/OnyxDialog/OnyxDialog.ct.tsx:47]: as I understand this rule, the idea is that a scroll container must be keyboard navigatable so you can use the arrow keys to scroll. so even if the project has a focusable button or so, if there is a long text, if you don't provide focus on the scroll area, the user won't be able to read all content.

so I'd not disable this rule here and instead make the scroll container focusable.

**Review by BoppLi — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/components/OnyxDialog/OnyxDialog.ct.tsx (hunk 1 above)
Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxDialog/OnyxDialog.ct.tsx:47]: [further info](https://dequeuniversity.com/rules/axe/4.8/scrollable-region-focusable)

> The key to getting a scrollable region to be accessible with the keyboard is to ensure that a keyboard-only user **can focus the scrollable region itself or a static text item within the scrollable region**. The first example below shows the first technique and the second one shows the second technique.

Comment (BoppLi) [reply packages/sit-onyx/src/components/OnyxDialog/OnyxDialog.ct.tsx:47]: uff that's a conflict... let's discuss it monday with the team how to proceed?

Comment (larsrickert) [reply packages/sit-onyx/src/components/OnyxDialog/OnyxDialog.ct.tsx:47]: Sure, why is this a conflict?

**Review by BoppLi — APPROVED** [review]

**Pull request merged**